### PR TITLE
Update Environment.OSVersion for browser platform

### PR DESCRIPTION
**Pull request by marek-safar**
Introduce PlatformID.Unknown for this is "legacy" API to allow returning some sensible value

Gathering quick feedback if this makes sense instead of e.g. PNSE

@lewing

### CODE REVIEW

File: src/libraries/System.Runtime/ref/System.Runtime.cs [hunk 1]
```diff
@@ -3072,6 +3072,7 @@ public enum PlatformID
         Unix = 4,
         Xbox = 5,
         MacOSX = 6,
+        Unknown = 7,
```
Comment (stephentoub) [inline src/libraries/System.Runtime/ref/System.Runtime.cs:3075]: Was this API reviewed?

Comment (marek-safar) [reply src/libraries/System.Runtime/ref/System.Runtime.cs:3075]: Not yet. Does it make sense to your?

Comment (stephentoub) [reply src/libraries/System.Runtime/ref/System.Runtime.cs:3075]: Seems ok.

**Comment by marek-safar** [comment]
Blocked by #38984 38984

File: src/libraries/System.Runtime/ref/System.Runtime.cs [hunk 2]
```diff
@@ -3072,6 +3072,7 @@ public enum PlatformID
         Unix = 4,
         Xbox = 5,
         MacOSX = 6,
+        Other = 7,
```
Comment (stephentoub) [inline src/libraries/System.Runtime/ref/System.Runtime.cs:3075]: Not related to this PR, but, @terrajobst, I see that the src has EditorBrowsableState.Never on some entries, but the ref doesn't have that.  Which is correct?

File: src/libraries/System.Private.CoreLib/src/System/Environment.OSVersion.Browser.cs [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System
+{
+    public static partial class Environment
+    {
+        private static OperatingSystem GetOSVersion()
+        {
+            return new OperatingSystem(PlatformID.Other, new Version(1, 0, 0, 0));
```
Comment (stephentoub) [inline src/libraries/System.Private.CoreLib/src/System/Environment.OSVersion.Browser.cs:11]: Can you add a test for this?

**Review by stephentoub — APPROVED** [review]

File: src/libraries/System.Runtime/ref/System.Runtime.cs (hunk 2 above)
Comment (stephentoub) [reply src/libraries/System.Runtime/ref/System.Runtime.cs:3075]: It looks like they're supposed to be hidden in the ref:
https://github.com/dotnet/runtime/issues/21660
https://github.com/dotnet/corefx/pull/19697
It looks like they were removed as part of:
https://github.com/dotnet/runtime/pull/31991/files#diff-1d02aa10f6540e37197ff9f090ca1d9eR2944
https://github.com/dotnet/runtime/pull/31991/files#diff-5450da078ed8d6b7e0e7413e10ab68c6L953.
@jkotas, was that intentional?

Comment (stephentoub) [reply src/libraries/System.Runtime/ref/System.Runtime.cs:3075]: Fixing in https://github.com/dotnet/runtime/pull/39065

Comment (jkotas) [reply src/libraries/System.Runtime/ref/System.Runtime.cs:3075]: It was not intentional. I guess I have just hit a GenAPI bug that I did not notice.

**Comment by marek-safar** [comment]
Reopening - it was closed by unrelated PR

File: src/libraries/System.Private.CoreLib/src/System/Environment.OSVersion.Browser.cs (hunk 3 above)
Comment (marek-safar) [reply src/libraries/System.Private.CoreLib/src/System/Environment.OSVersion.Browser.cs:11]: There are tests already which are being fixed/enabled as part of #38996

**Pull request merged**